tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html

```diff
@@ -4266,6 +4287,43 @@ <h2>Show similarity to selected datapoint</h2>
         this.updatedExample = false;
       },
 
+      newAttributions_: function(attributions) {
+        if (Object.keys(attributions).length == 0) {
+          return;
+        }
+        for (let i = 0; i < attributions.indices.length; i++) {
+          const idx = attributions.indices[i];
+          const datapoint = this.visdata[idx];
+          // TODO(jwexler): Support attributions from multiple models.
+          // For now, we only display attribution from the first model, if WIT
+          // is loaded with two models.
+          const attribs = attributions.attributions[0][i];
+          const keys = Object.keys(attribs);
+          for (let j = 0; j < keys.length; j++) {
+            // If the attributions for a key is a 2D array then treat the first
+            // entry as sparse keys for attributions and the second entry as
+            // the attributions for the sparse keys in the first array.
+            // Otherwise the attribution is a single value or list and can be
+            // used as such.
+            if (Array.isArray(attribs[keys[j]]) &&
+                attribs[keys[j]].length == 2 &&
+                Array.isArray(attribs[keys[j]][0])) {
+              const attribSparseKeyKey = attributionPrefix + keys[j] +
+                attributionSparseKeySuffix;
+              datapoint[attribSparseKeyKey] = attribs[keys[j]][0];
+              const attribSparseValueKey = attributionPrefix + keys[j] +
+                attributionSparseValueSuffix;
+              datapoint[attribSparseValueKey] = attribs[keys[j]][1];
+            } else {
+              const attribKey = attributionPrefix + keys[j];
+              datapoint[attribKey] = attribs[keys[j]];
```

Comment:
it feels un-Polymermic to mutate data like this and then trigger a draw (as opposed to using prop changes to trigger data binding). Can you consider creating another object and do ```this.set(`visdata[${index}]`, newObject)```?

Reply:
fixing